std/internal/math/biguintcore.d

```diff
@@ -395,26 +408,30 @@ public:
         return true;
     }
 
-    // return true if OK; false if erroneous characters found
-    // FIXME: actually throws `ConvException` on error.
-    bool fromDecimalString(const(char)[] s) pure @trusted
+    /// return true if OK; false if erroneous characters found
+    bool fromDecimalString(Range)(Range s) if (
+        isForwardRange!Range && isSomeChar!(ElementType!Range))
     {
+        import std.range : walkLength;
+        import std.stdio;
+
         //Strip leading zeros
-        int firstNonZero = 0;
-        while ((firstNonZero < s.length) &&
-            (s[firstNonZero]=='0' || s[firstNonZero]=='_'))
+        size_t first_non_zero;
+        while (!s.empty && (s.front == '0' || s.front == '_'))
         {
-                ++firstNonZero;
+            ++first_non_zero;
+            s.popFront;
         }
-        if (firstNonZero == s.length && s.length >= 1)
+
+        if (s.empty)
         {
             data = ZERO;
             return true;
         }
-        auto predictlength = (18*2 + 2*(s.length-firstNonZero)) / 19;
```

Comment:
This line has the same bug. Unfortunately no unittest triggers the problem. (Add one?)